parser-typechecker/unison/Main.hs

```diff
@@ -63,60 +63,71 @@ usage = P.callout "🌻" $ P.lines [
 
 main :: IO ()
 main = do
-  args               <- getArgs
+  args <- getArgs
   -- hSetBuffering stdout NoBuffering -- cool
-  (codepath, args) <- case args of
-    "-codebase" : codepath : args -> pure (codepath, args)
-    _ -> (,) <$> getHomeDirectory <*> pure args
+
+  -- We need to know whether the program was invoked with -codebase for
+  -- certain messages. Therefore we keep a Maybe FilePath - mcodepath
+  -- rather than just deciding on whether to use the supplied path or
+  -- the home directory here and throwing away that bit of information
```

Comment:
Thanks for including this comment.  I was about to ask, and then I thought "what if I try reading..."